homeassistant/components/sensor/starlingbank.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for balance data via the Starling Bank API.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/components/sensor.starlingbank/
+"""
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN, CONF_NAME
+)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['starlingbank==1.1']
+
+BALANCE_TYPES = ['cleared_balance', 'effective_balance']
+DEFAULT_ACCOUNT_NAME = 'Starling'
+ICON = 'mdi:currency-gbp'
+
+CONF_ACCOUNTS = 'accounts'
+CONF_BALANCE_TYPES = 'balance_types'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ACCOUNTS): vol.All([{
+            vol.Optional(CONF_NAME, default=DEFAULT_ACCOUNT_NAME): cv.string,
+            vol.Required(CONF_ACCESS_TOKEN): cv.string,
+            vol.Optional(
+                CONF_BALANCE_TYPES, default=BALANCE_TYPES
+                ): vol.All(
+                    cv.ensure_list,  # Wraps value in a list if it is not one.
+                    [vol.In(BALANCE_TYPES)]
+                )
+        }], cv.ensure_list)
+    }
+)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Sensor platform setup."""
+    accounts = config.get(CONF_ACCOUNTS)
+
+    sensors = []
+    for account in accounts:
+        for balance_type in account[CONF_BALANCE_TYPES]:
+            sensors.append(StarlingBalanceSensor(
+                account[CONF_ACCESS_TOKEN],
+                account[CONF_NAME],
+                balance_type
+            ))
+    add_devices(sensors, True)  # True forces an update upon platform setup.
```

Comment:
You should check that the Access token is valid during the setup of the platform. If not then fail and log the error. Otherwise will be users end up with a non-working platform in their instance.

Reply:
@fabaff - Thanks for the feedback. I have made some changes to catch and log HTTP exceptions (which would be raised if the token is rejected). Is the change I've made sufficient?